### PR TITLE
Combine lora msg

### PR DESCRIPTION
**Pull request by nerdyscout**
this pull request will enable to send gps data on same port as counter

### CODE REVIEW

**Comment by nerdyscout** [comment]
don't know why you already merged this, told you there might be more changes.

**Comment by cyberman54** [comment]
Is this PR still work in progress?
If not, yould you please detach the "draft" status (otherwise it can't be merged) and rebase it to development branch? Thanks.

File: src/gpsread.cpp [hunk 1]
```diff
@@ -76,9 +76,11 @@ void gps_storelocation(gpsStatus_t *gps_store) {
       (gps.location.age() < 1500)) {
     gps_store->latitude = (int32_t)(gps.location.lat() * 1e6);
     gps_store->longitude = (int32_t)(gps.location.lng() * 1e6);
+#if !(PAYLOAD_OPENSENSEBOX)
     gps_store->satellites = (uint8_t)gps.satellites.value();
```
Comment (cyberman54) [inline src/gpsread.cpp]: We need satellites value for display, also if we do not want opensensebox payload.
Better leave the struct untouched by the compile flag.

File: include/globals.h [hunk 2]
```diff
@@ -95,9 +95,11 @@ typedef struct {
 typedef struct {
   int32_t latitude;
   int32_t longitude;
+#if !(PAYLOAD_OPENSENSEBOX)
   uint8_t satellites;
```
Comment (cyberman54) [inline include/globals.h]: Suggest to leave this untouched, doesn't hurt.

File: src/senddata.cpp [hunk 3]
```diff
@@ -65,12 +65,12 @@ void sendData() {
 #if ((WIFICOUNTER) || (BLECOUNTER))
     case COUNT_DATA:
       payload.reset();
-      if (cfg.wifiscan)
-        payload.addCount(macs_wifi, MAC_SNIFF_WIFI);
+#if !(PAYLOAD_OPENSENSEBOX)
+      payload.addCount(macs_wifi, MAC_SNIFF_WIFI);
```
Comment (cyberman54) [inline src/senddata.cpp]: add
 if (cfg.wifiscan)

File: src/senddata.cpp [hunk 4]
```diff
@@ -79,7 +79,11 @@ void sendData() {
       } else
         ESP_LOGD(TAG, "No valid GPS position");
 #endif
-
+#if (PAYLOAD_OPENSENSEBOX)      
+      payload.addCount(macs_wifi, MAC_SNIFF_WIFI);
```
Comment (cyberman54) [inline src/senddata.cpp]: add
 if (cfg.wifiscan)

**Comment by nerdyscout** [comment]
openSenseBox decoding requires sending only latitude and longitude, therefore modification of the gps payload is needed. With the current built i can not get a gps fix anymore, can't figure out why... everything else should be fine. but i will be off for a few days now.

**Comment by cyberman54** [comment]
We just need to adjust the payload construction in payload.cpp, as you did it, but no need to touch the internal struct containing the (live) gps data.

**Comment by nerdyscout** [comment]
everything should be fine now.
I guess it fixes #498 as well.

File: src/senddata.cpp (hunk 3 above)
Comment (nerdyscout) [reply src/senddata.cpp]: done

File: src/senddata.cpp (hunk 4 above)
Comment (nerdyscout) [reply src/senddata.cpp]: done

File: include/globals.h (hunk 2 above)
Comment (nerdyscout) [reply include/globals.h]: reworked

**Pull request merged**